torch/distributed/distributed_c10d.py

```diff
@@ -461,8 +471,12 @@ class GroupMember(metaclass=_WorldMeta):
 def _get_pg_device(group: ProcessGroup):
     """
     Returns the device to use with ``group``.
-    This is cuda for NCCL and CPU for everything else
+    While group isn't ProcessGroup, this is cuda for NCCL and CPU for everything else
     """
+    if isinstance(group, ProcessGroup):
```

Comment:
Given the typing annotation above, this doesn't really make sense? This will always be True?

Reply:
This will not always be True. Currently, class `ProcessGroupGloo`, `ProcessGroupNccl` is subclass of `Backend`, not subclass of `ProcessGroup`. e.g.`isinstance(ProcessGroupGloo, ProcessGroup) == False`. This change is primarily for the `ProcessGroup` that will initialize by user.

Reply:
But then the typing for this function is wrong?
cc @awgu

Reply:
@H-Huang I thought ``Backend`` was not exposed to python.

@huihoaan do you have an example of a codepath that exposes this?